src/read_pair_writer.rs

```diff
@@ -47,9 +47,9 @@ impl ReadPairWriter {
         let mut writers = [None, None, None, None];
         let mut paths = [None, None, None, None];
 
-        for (idx, r) in [r1, r2, i1, i2].iter().enumerate() {
-            if let Some(ref p) = *r {
-                let wtr = utils::write_with_gz(p)?;
+        for (idx, r) in IntoIterator::into_iter([r1, r2, i1, i2]).enumerate() {
```

Comment:
This should work?
```suggestion
        for (idx, r) in [r1, r2, i1, i2].into_iter().enumerate() {
```

Reply:
Unfortunately no, because [the meaning of `.into_iter()` on an array changes depending on whether you're using 2018 or 2021 edition](https://doc.rust-lang.org/edition-guide/rust-2021/IntoIterator-for-arrays.html), which triggers a migration lint warning if you're still on the 2018 edition (which we are).

Reply:
Ah okay!